src/main/java/edu/cornell/kfs/fp/batch/service/impl/InternalBillingDocumentGenerator.java

```diff
@@ -0,0 +1,63 @@
+package edu.cornell.kfs.fp.batch.service.impl;
+
+import java.sql.Timestamp;
+import java.util.function.Supplier;
+
+import org.kuali.kfs.fp.businessobject.InternalBillingItem;
+import org.kuali.kfs.fp.document.InternalBillingDocument;
+import org.kuali.kfs.krad.bo.AdHocRoutePerson;
+import org.kuali.kfs.krad.bo.Note;
+
+import edu.cornell.kfs.fp.batch.xml.AccountingXmlDocumentEntry;
+import edu.cornell.kfs.fp.batch.xml.AccountingXmlDocumentItem;
+
+public class InternalBillingDocumentGenerator
+        extends AccountingDocumentGeneratorBase<InternalBillingDocument> {
+
+    public InternalBillingDocumentGenerator() {
+        super();
+    }
+
+    public InternalBillingDocumentGenerator(
+            Supplier<Note> emptyNoteGenerator, Supplier<AdHocRoutePerson> emptyAdHocRoutePersonGenerator) {
+        super(emptyNoteGenerator, emptyAdHocRoutePersonGenerator);

```

Comment:
since these constructors are just calling their super, are they needed here?

Reply:
Yes, these are needed so that this class knows which superclass constructor to call in each case, and to allow for properly invoking the two-arg constructor when needed. Without an explicit constructor, only the superclass's no-arg constructor (if available and visible) will be used when invoking the implicit no-arg constructor on this class.

Reply:
OK, thanks for clarifying